lib/utils.js

```diff
@@ -67,7 +66,14 @@ function normalizeBinary (binaryPath, platform, arch) {
     // On linux but no path yet, use which to try and find the binary
     else if (platform.indexOf("linux") !== -1) {
         binaryPath = normalizeBinary.appNames[binaryPath + " on linux"] || binaryPath;
-        return resolve(which(binaryPath));
+        which(binaryPath, function(err, resolvedPath) {
```

Comment:
It looks strange to replace return with not returning, but upon looking at the context it is obvious that the function body is part of a Promise executor, for which the return value is not significant. Only resolve/reject calls (and synchronously thrown errors) matter.